js/tab.js

```diff
@@ -171,6 +180,32 @@ define([
     return syntaxValue;
   }
   
+  Tab.prototype.getWorker = function() {
+    return new Promise((resolve, reject) => {
+      // if worker is off, return null
+      if (!Settings.get("user").useWorker) return resolve(null);
+      
+      var checkLimit = 20;
+      var findWorker = () => {
+        // if worker was turned while we were waiting on it, return null
+        if (!Settings.get("user").useWorker) return resolve(null);
+      
+        // return the worker if we found it
+        if (this.$worker) {
+          return resolve(this.$worker);
+        }
+        
+        // see if we've checked too many times
+        if (checkLimit-- < 0) return reject(new Error('Could not find worker.'));
+        
+        // otherwise, go to sleep and then check again
+        setTimeout(findWorker, 100);
```

Comment:
Well that's... wildly unfortunate.

I was going to suggest using Object.observe, but it turns out that's been deprecated in favor of Proxies, which seem like overkill for this purpose. So I have a suggestion: rather than set the JSHint config from the same place as the rest of the options, let's override `$startWorker` (since tabs are just a subclass of EditSession). Maybe something like this?

```js
$startWorker: function() {
  //call the superclass method
  EditSession.prototype.$startWorker.call(this);
  if (this.syntax == "javascript" && this.$worker) {
    //send your message to the worker
  }
}
```